### ISSUE TITLE
Kubernetes v2 namespace account restrictions ignored

### ISSUE THREAD

**Issue by battlecow** [spinnaker/spinnaker#5769]
### Issue Summary:
We are currently restricting accounts for kubernetes config to allow deployments into only certain namespaces.  While I am fairly certain this worked in 1.18.x, using 1.19.8 the pipeline can be used to deploy to a namespace outside of the specified restrictions.  We are using external accounts via spring config server and loading all of our `clouddriver.yaml` and `kubeconfig` from our internal git server.

### Cloud Provider(s):
Kubernetes v2

### Steps to Reproduce:
Create clouddriver config with namespace restrictions for a kubernetes account e.g.:
```
 - name: "services-cicd-use1"
    requiredGroupMembership: []
    providerVersion: V2
    permissions:
      READ:
      - team
      WRITE:
      - team
      EXECUTE:
      - team
    dockerRegistries: []
    context: "services-cicd-use1"
    configureImagePullSecrets: true
    cacheThreads: 1
    namespace: ["cicd"]
    omitNamespaces: []
    kinds: []
    omitKinds:
    - podPreset
    customResources: []
    cachingPolicies: []
    kubeconfigFile: configserver:kubeconfig.yml
    liveManifestCalls: true
    oAuthScopes: []
    onlySpinnakerManaged: true
```
Reference the kubeconfig:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: "xxx"
    server: "http://xxxx"
  name: "services-cicd-use1"
contexts:
- context:
    cluster: "services-cicd-use1"
    user: "services-cicd-use1"
  name: "services-cicd-use1"
kind: Config
preferences: {}
users:
- name: "services-cicd-use1"
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - token
      - -i
      - "cicd-use1"
      - -r
      - "arn:aws:iam::1111111111:role/spinnakerManaged"
      command: aws-iam-authenticator
```

Create a pipeline to deploy a manifest to a namespace outside of the `cicd` namespace:
```
{
  "isNew": true,
  "keepWaitingPipelines": false,
  "limitConcurrent": true,
  "spelEvaluator": "v4",
  "stages": [
    {
      "account": "services-cicd-use1",
      "cloudProvider": "kubernetes",
      "isNew": true,
      "manifests": [
        {
          "apiVersion": "v1",
          "kind": "Pod",
          "metadata": {
            "labels": {
              "run": "alpine"
            },
            "name": "alpine",
            "namespace": "default"
          },
          "spec": {
            "containers": [
              {
                "args": [
                  "/bin/sh"
                ],
                "image": "alpine",
                "imagePullPolicy": "Always",
                "name": "alpine",
                "resources": {},
                "stdin": true,
                "stdinOnce": true,
                "terminationMessagePath": "/dev/termination-log",
                "terminationMessagePolicy": "File",
                "tty": true
              }
            ]
          }
        }
      ],
      "moniker": {
        "app": "test"
      },
      "name": "Deploy (Manifest)",
      "namespaceOverride": "",
      "refId": "1",
      "requisiteStageRefIds": [],
      "skipExpressionEvaluation": false,
      "source": "text",
      "trafficManagement": {
        "enabled": false,
        "options": {
          "enableTraffic": false,
          "namespace": null,
          "services": [],
          "strategy": null
        }
      },
      "type": "deployManifest"
    }
  ],
  "triggers": [],
  "updateTs": "1589488953706"
}
```

Expectation: It should fail
Actual result: It deploys the manifest

**Comment by ezimanyi** [comment]
@battlecow : Thanks for opening this!  I just tried to reproduce this myself and got the following when trying to deploy to a namespace (which I cleverly named `no-access`) that was not in my explicit list of `namespaces` in my config:
![image](https://user-images.githubusercontent.com/1391982/83180313-40073c80-a0f1-11ea-930b-9a93ae8dd462.png)

After testing that and being confused, I took a closer look at your config above...it looks like in you've specified the namespace as a singular `namespace: ["cicd"]`.  This should be plural as `namespaces: ["cicd"]`.  I suspect that's the issue you're seeing.
